cpp/src/strings/utilities.cu

```diff
@@ -140,6 +143,28 @@ std::unique_ptr<column> make_empty_strings_column( rmm::mr::device_memory_resour
                                      rmm::device_buffer{0,stream,mr}, 0 ); // nulls
 }
 
+namespace
+{
+
+// This device variable is created here to avoid using a singleton that may cause issues
+// with RMM initialize/finalize. See PR #3159 for details on this approach.
+__device__ character_flags_table_type character_codepoint_flags[sizeof(g_character_codepoint_flags)];
```

Comment:
Wouldn't it be easier to just make `g_character_codepoint_flags` a `__device__` array and avoid the need for a `cudaMemcpyToSymbol` all of this additional logic (mutex, cudaMemcpyToSymbol, etc.).

Reply:
I could not get that to work. I need to use this table in many places and do not want to include it everywhere. I just want it included here and just return a pointer everywhere it can be used.